flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecTableSourceScan.java

```diff
@@ -53,45 +52,41 @@ public class BatchExecTableSourceScan extends CommonExecTableSourceScan
         implements BatchExecNode<RowData> {
 
     // Avoids creating different ids if translated multiple times
-    private final String dynamicFilteringDataListenerID = UUID.randomUUID().toString();
+    @Nullable private final String dynamicFilteringDataListenerID;
 
-    private boolean needDynamicFilteringDependency;
+    private final ReadableConfig tableConfig;
 
     // This constructor can be used only when table source scan has
     // BatchExecDynamicFilteringDataCollector input
     public BatchExecTableSourceScan(
             ReadableConfig tableConfig,
             DynamicTableSourceSpec tableSourceSpec,
-            InputProperty inputProperty,
+            List<InputProperty> inputProperties,
             RowType outputType,
-            String description) {
+            String description,
+            @Nullable String dynamicFilteringDataListenerID) {
         super(
                 ExecNodeContext.newNodeId(),
                 ExecNodeContext.newContext(BatchExecTableSourceScan.class),
                 ExecNodeContext.newPersistedConfig(BatchExecTableSourceScan.class, tableConfig),
                 tableSourceSpec,
-                Collections.singletonList(inputProperty),
+                inputProperties,
                 outputType,
                 description);
+        this.dynamicFilteringDataListenerID = dynamicFilteringDataListenerID;
+        this.tableConfig = tableConfig;
     }
 
     public BatchExecTableSourceScan(
             ReadableConfig tableConfig,
             DynamicTableSourceSpec tableSourceSpec,
             RowType outputType,
             String description) {
-        super(
-                ExecNodeContext.newNodeId(),
-                ExecNodeContext.newContext(BatchExecTableSourceScan.class),
-                ExecNodeContext.newPersistedConfig(BatchExecTableSourceScan.class, tableConfig),
-                tableSourceSpec,
-                Collections.emptyList(),
-                outputType,
-                description);
+        this(tableConfig, tableSourceSpec, Collections.emptyList(), outputType, description, null);
```

Comment:
I think the original constructor can work for our needs, just needs to do a small enforcement.
```
    public BatchExecTableSourceScan(
            ReadableConfig tableConfig,
            DynamicTableSourceSpec tableSourceSpec,
            RowType outputType,
            String description) {
        super(
                ExecNodeContext.newNodeId(),
                ExecNodeContext.newContext(BatchExecTableSourceScan.class),
                ExecNodeContext.newPersistedConfig(BatchExecTableSourceScan.class, tableConfig),
                tableSourceSpec,
                Collections.emptyList(),
                outputType,
                description);
        this.tableConfig = tableConfig;
    }
```

Reply:
Fixed